nexus/src/db/lookup.rs

```diff
@@ -231,6 +231,23 @@ impl<'a> LookupPath<'a> {
         Disk { key: DiskKey::PrimaryKey(Root { lookup_root: self }, id) }
     }
 
+    /// Select a resource of type Image, identified by its id
+    pub fn image_id(self, id: Uuid) -> Image<'a> {
+        Image { key: ImageKey::PrimaryKey(Root { lookup_root: self }, id) }
+    }
+
+    /// Select a resource of type Snapshot, identified by its id
+    pub fn snapshot_id(self, id: Uuid) -> Snapshot<'a> {
+        Snapshot { key: SnapshotKey::PrimaryKey(Root { lookup_root: self }, id) }
+    }
```

Comment:
Snapshots aren't technically real at this point. I can remove all snapshot related changes if others think it would be prudent to do so.

Reply:
Keep them in, they'll be real enough soon :)